packages/@react-spectrum/dialog/docs/DialogTrigger.mdx

```diff
@@ -0,0 +1,461 @@
+portion of the page behind the underlay. Use this variant for more complex workflow that
+do not fit on the available Modal Dialog sizes.
+
+```tsx example
+<DialogTrigger type="fullscreen">
+  <ActionButton>See Details</ActionButton>
+  {(close) => ( 
+    <Dialog>
+      <Heading>Package details</Heading>
+      <Divider />
+      <Content>
+        <Text>
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin sit amet tristique risus. In sit amet suscipit lorem. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. In condimentum imperdiet metus non condimentum. Duis eu velit et quam accumsan tempus at id velit. Duis elementum elementum purus, id tempus mauris posuere a. Nunc vestibulum sapien pellentesque lectus commodo ornare.
+        </Text>
+      </Content>
+      <ButtonGroup>
+        <Button variant="secondary" onPress={close}>Cancel</Button>
+        <Button variant="cta" onPress={close} autoFocus>Buy</Button>
+      </ButtonGroup>    
+    </Dialog>
+  )}
+</DialogTrigger>
+```
+
+### Fullscreen Takeover
+
+Fullscreen takeover Dialogs are similar to the fullscreen variant except that the 
+background covers the entire screen. This variant should be reserved for workflows 
+where displaying a second dialog on top of the first one is to be expected.
+
+```tsx example
+<DialogTrigger type="fullscreenTakeover">
+  <ActionButton>Register</ActionButton>
+  {(close) => ( 
+    <Dialog>
+      <Heading>Register a new account</Heading>
+      <Divider />
+      <Content>
+        <Form>
+          <TextField label="Name" />
+          <TextField label="Email address" />
+          <Checkbox>Make profile private</Checkbox>
+        </Form>
+        <DialogTrigger isDismissable type="modal">
+          <ActionButton>Privacy Policy</ActionButton>
+          <Dialog>
+            <Heading>Privacy Policy</Heading>
+            <Divider />
+            <Content>
+              <Text>
+                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin sit amet tristique risus. In sit amet suscipit lorem. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. In condimentum imperdiet metus non condimentum. Duis eu velit et quam accumsan tempus at id velit. Duis elementum elementum purus, id tempus mauris posuere a. Nunc vestibulum sapien pellentesque lectus commodo ornare.
+              </Text>
+            </Content>
+          </Dialog>
+        </DialogTrigger>
+      </Content>
+      <ButtonGroup>
+        <Button variant="secondary" onPress={close}>Cancel</Button>
+        <Button variant="cta" onPress={close} autoFocus>Confirm</Button>
+      </ButtonGroup>    
+    </Dialog>
+  )}
+</DialogTrigger>
+```
+
+### Dismissable
+
+If your Modal Dialog doesn't require the user to make a confirmation, you can set `isDismissable` 
+on the DialogTrigger. This adds an exit button that the user can click to dismiss the Dialog.
```

Comment:
it also enables the `esc` key

Reply:
I'll mention this under the `Dialog Types` section since the `esc` key applies to all dialog types